### PR TITLE
F/fix partition python

### PR DESCRIPTION
**Pull request by dsudduth**
*Issue #, if available:*
#332

*Description of changes:*
This Pull Request adds support for GovCloud customers and regions. As described in #332, the changes focus on dynamically setting the partition.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by StewartW** [comment]
This looks good @dsudduth . 
Thanks for the contribution! 

Would you be able to rebase it into a single (or fewer) commits though please?

**Comment by dsudduth** [comment]
> This looks good @dsudduth .
> 
> I guess it makes: #343 redundant?
> Thanks for the contribution!
> 
> Would you be able to rebase it into a single (or fewer) commits though please?

Hi @StewartW,

I agree, #343 would no longer be needed. I've squashed the changes down significantly, keeping a few. Let me know if we need to change anything else. Thanks again for the review!

**Review by sbkok — CHANGES_REQUESTED** [review]

File: src/lambda_codebase/cross_region_bucket/main.py [hunk 1]
```diff
@@ -143,19 +145,23 @@ def ensure_bucket(region: str, bucket_name_prefix: str) -> Tuple[BucketName, Cre
...
                 **config
             )
-            LOGGER.info("Bucket created: %s", bucket_name)
+            LOGGER.info(f"Bucket created: {bucket_name}")
             return bucket_name, True
         except s3_client.exceptions.BucketAlreadyExists:
             LOGGER.info(
-                "Bucket name %s already taken, trying another one ...", bucket_name
+                f"Bucket name {bucket_name} already taken, "
+                f"trying another one ..."
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/cross_region_bucket/main.py]: Updated in latest commit.

File: src/lambda_codebase/cross_region_bucket/main.py [hunk 2]
```diff
@@ -185,11 +191,14 @@ def ensure_bucket_has_no_public_access(bucket_name: str, region: str) -> None:
 
 
 def ensure_bucket_policy(bucket_name: str, region: str, policy: MutableMapping) -> None:
+
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/cross_region_bucket/main.py]: Updated in latest commit.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 3]
```diff
@@ -32,14 +32,14 @@ Parameters:
     Type: String
     Default: "BUILD_GENERAL1_LARGE" # For threading with large amounts of pipelines this is the most effective default
     AllowedValues:
-      - "BUILD_GENERAL1_SMALL"  #3 GB memory, 2 vCPU
+      - "BUILD_GENERAL1_SMALL" #3 GB memory, 2 vCPU
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml]: Updated in latest commit.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 4]
```diff
@@ -1205,35 +1205,56 @@ Resources:
...
+                    "",
+                    [
+                      !Sub "arn:${AWS::Partition}:codepipeline:",
+                      !Ref "AWS::Region",
+                      ":",
+                      !Ref "AWS::AccountId",
+                      ":",
+                      !Ref CodePipeline,
+                    ],
+                  ]
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml]: Updated in latest commit.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 5]
```diff
@@ -1205,35 +1205,56 @@ Resources:
...
+              "",
+              [
+                !Sub "arn:${AWS::Partition}:codecommit:",
+                !Ref "AWS::Region",
+                ":",
+                !Ref "AWS::AccountId",
+                ":",
+                !GetAtt CodeCommitRepository.Name,
+              ],
+            ]
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml]: Updated in latest commit.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 6]
```diff
@@ -1244,7 +1265,17 @@ Resources:
...
+              "",
+              [
+                !Sub "arn:${AWS::Partition}:codepipeline:",
+                !Ref "AWS::Region",
+                ":",
+                !Ref "AWS::AccountId",
+                ":",
+                !Ref CodePipeline,
+              ],
+            ]
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml]: Updated in latest commit.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_constructs/adf_codepipeline.py [hunk 7]
```diff
@@ -258,8 +275,11 @@ def _generate_configuration(self): #pylint: disable=R0912, R0911, R0915
         raise Exception("{0} is not a valid provider".format(self.provider))
 
     def _generate_codepipeline_access_role(self): #pylint: disable=R0911
+
```
[the remark this answers is not in the document]
Comment (dsudduth) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_constructs/adf_codepipeline.py]: Updated in latest commit.

**Review by sbkok — APPROVED** [review]
Thanks a lot for following up with these changes so quickly.
And thank you for contributing, looks great to me!

**Review by StewartW — APPROVED** [review]
:shipit:  Changes look good. Thanks for making them

**Pull request merged**